### PR TITLE
Followed the official Twitter API procedure (Multipart Media Upload)

### PR DESCRIPTION
**Pull request by Kyome22**
I'm sorry to have troubled you through my oversight.
- I changed that  STATUS command is called after the FINALIZE command.
- I changed that success method is run if a sync response comes back after the FINALIZE command instead of an async response. ← I hope this modification resolves issue #313.

See: https://developer.twitter.com/en/docs/media/upload-media/api-reference/post-media-upload-finalize

### CODE REVIEW

File: Sources/SwifterMedia.swift [hunk 1]
```diff
@@ -60,22 +71,56 @@ public extension Swifter {
         let parameters = ["command": "FINALIZE",
                           "media_id": mediaId]
         self.postJSON(path: path, baseURL: .upload, parameters: parameters, success: { (json, response) in
+            guard let obj = json.object, obj.keys.count > 0 else {
+                let error = SwifterError(message: "Bad Response for Multipart Media Upload",
+                                         kind: .invalidMultipartMediaResponse)
+                failure?(error)
+                return
+            }
+            if let processingInfo = json["processing_info"].object, let state = processingInfo["state"]?.string {
```
Comment (meteochu) [inline Sources/SwifterMedia.swift:80]: This looks very similar to the response handler in `statusUpload(mediaId:success:failure:)`, can you wrap this into a separate private helper method so we don't have code duplication here?

File: Sources/SwifterMedia.swift [hunk 2]
```diff
@@ -60,22 +71,56 @@ public extension Swifter {
...
+                // async FINALIZE response which requires further STATUS command call(s)
+                switch state {
+                case "pending", "in_progress":
+                    let secs = processingInfo["check_after_secs"]?.double ?? 3.0
```
Comment (meteochu) [inline Sources/SwifterMedia.swift:84]: 3 seconds seem like a really long time to retry. Can we not guarantee from the API that `check_after_secs` will always exist?

Comment (Kyome22) [reply Sources/SwifterMedia.swift:84]: Perhaps we can think that check_after_secs always returns at the time of pending and in_progress.

File: Sources/SwifterMedia.swift (hunk 1 above)
Comment (Kyome22) [reply Sources/SwifterMedia.swift]: 2b53ea2
Modified my code.

**Comment by meteochu** [comment]
Awesome, this LGTM 👍.

**Pull request merged**